client/orderbook/bookside.go

```diff
@@ -216,14 +230,13 @@ func (d *bookSide) idxCalculator() func(i int) int {
 }
 
 func (d *bookSide) iterateOrders(check func(*Order) bool) {
-	calcIdx := d.idxCalculator()
-
-	for i := 0; i < len(d.rateIndex.Rates); i++ {
-		bin := d.bins[d.rateIndex.Rates[calcIdx(i)]]
+	calcIdx := d.idxCalculator() // ascending or descending Rates index
 
-		for idx := 0; idx < len(bin); idx++ {
-			if !check(bin[idx]) {
-				break
+	for ir := range d.rateIndex.Rates {
+		bin := d.bins[d.rateIndex.Rates[calcIdx(ir)]]
+		for _, ord := range bin {
+			if !check(ord) {
+				break // really next rate bin? or break outer / return?
```

Comment:
Oh, I wasn't sure if this was a bug.  Seems like if `check` returns `false`, the iterations should stop altogether.  This `break` just breaks the *inner* for loop.

Reply:
Looks like it should just `return`

Reply:
Changed to `return`.  Pretty certain the idea is that there's no reason to check other bins if check says false for an order in current bin?